galaxy_ng/app/api/v1/viewsets.py

```diff
@@ -0,0 +1,358 @@
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import AllowAny
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.exceptions import ValidationError
+
+from pulpcore.plugin.models import Task
+from pulpcore.plugin.tasking import dispatch
+
+from galaxy_ng.app.api.v1.tasks import legacy_role_import
+from galaxy_ng.app.api.v1.tasks import legacy_sync_from_upstream
+from galaxy_ng.app.api.v1.models import LegacyNamespace
+from galaxy_ng.app.api.v1.models import LegacyRole
+from galaxy_ng.app.api.v1.serializers import LegacyRoleSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyRoleContentSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyRoleVersionsSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyUserSerializer
+
+
+logger = logging.getLogger(__name__)
+
+
+class LegacyUserSetPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+
+class LegacyUserViewSet(viewsets.ModelViewSet):
+    queryset = LegacyNamespace.objects.all().order_by('name')
+    serializer = LegacyUserSerializer
+    serializer_class = LegacyUserSerializer
+    permission_classes = [AllowAny]
+    pagination_class = LegacyUserSetPagination
+
+    def get_user(self, *args, **kwargs):
+        userid = kwargs.get('userid')
+        user = LegacyNamespace.objects.filter(id=userid).first()
+        serializer = LegacyUserSerializer(user)
+        return Response(serializer.data)
+
+    def get_queryset(self):
+
+        logger.info(f'QUERY_PARAMS: {self.request.query_params}')
+
+        keywords = None
+        if self.request.query_params.get('keywords'):
+            keywords = self.request.query_params.get('keywords').rstrip('/')
+
+        order_by = 'name'
+        if self.request.query_params.get('order_by'):
+            order_by = self.request.query_params.get('order_by').rstrip('/')
+
+        if self.request.query_params.get('name'):
+            name = self.request.query_params.get('name').rstrip('/')
+            return LegacyNamespace.objects.filter(name=name).order_by(order_by)
+
+        if keywords:
+            return LegacyNamespace.objects.filter(
+                Q(name__contains=keywords)
+            ).order_by(order_by)
+
+        return LegacyNamespace.objects.all().order_by(order_by)
+
+
+class LegacyRolesSetPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+
+class LegacyRoleViewSet(viewsets.ModelViewSet):
+    queryset = LegacyRole.objects.all().order_by('full_metadata__created')
+    serializer = LegacyRoleSerializer
+    serializer_class = LegacyRoleSerializer
+    permission_classes = [AllowAny]
+    pagination_class = LegacyRolesSetPagination
+    authentication_classes = [BasicAuthentication, SessionAuthentication, TokenAuthentication]
+
+    def get_queryset(self):
```

Comment:
You might be able to clean up all this manual filtering by using a filterset (example: https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/api/ui/viewsets/collection.py#L155)

Reply:
done